site/public/js/ta-grading-rubric.js

```diff
@@ -1832,11 +1832,30 @@ function onClickComponent(me) {
  * @param me DOM Element of the cancel button
  */
 function onCancelComponent(me) {
-    toggleComponent(getComponentIdFromDOMElement(me), false)
-        .catch(function (err) {
+  const component_id = getComponentIdFromDOMElement(me);
+  const gradeable_id = getGradeableId();
+  const anon_id = getAnonId();
+  ajaxGetGradedComponent(gradeable_id, component_id, anon_id).then((component)=>{
+    // If there is any changes made in comment of a component , prompt the TA
+    if ( component.comment !== $('#component-' + component_id).find('.mark-note-custom').val().trim()) {
+      if(confirm("Note: Changes made to the Student Message will be lost! Do you want to continue ? ")){
```

Comment:
This warning message should be more direct,
something like "Are you sure you want to discard all changes to the student message?"

Reply:
done!